src/test/java/org/opensearch/neuralsearch/processor/ScoreCombinationTechniqueTests.java

```diff
@@ -27,7 +27,7 @@ public void testCombination_whenMultipleSubqueriesResultsAndDefaultMethod_thenSc
 
         final List<CompoundTopDocs> queryTopDocs = List.of(
             new CompoundTopDocs(
-                new TotalHits(3, TotalHits.Relation.EQUAL_TO),
+                new TotalHits(5, TotalHits.Relation.EQUAL_TO),
```

Comment:
Changing the count to 5 as this value will come from Query Phase. And in the data there are 5 unqiue scoredocs.

Reply:
Is this the only place in tests where we the check for totalHits was not correct? We may need to add more tests to check edge cases with totalHits

Reply:
Yes adding more.